misc/roundtrip

```diff
@@ -126,23 +126,20 @@ IFS='
 
 tagfiles=$(find "$UNITS" -name expected.tags)
 for tags in $tagfiles; do
-    tagnames=$(sed -e 's/^\([^	]*\)	.*/\1/' "$tags" | limit_entries)
-    for name in $tagnames; do
-	# Yes, there is a reason for this craziness.  We need to properly
-	# handle embedded newlines (expanded from "\n"), including trailing
-	# ones the shell would strip automatically.  To work around this, we
-	# add a dummy character at the end to inhibit stripping, and then
-	# remove it, plus the extra newline, using variable substitutions.
-	# Note: we use "printf '%s\n'" instead of "echo" because Dash's "echo"
-	# unconditionally expands some sequences, like "\t" and alike.
-	t="$(printf '%s\n' "$name" | expandEscapeSequences; printf _)"
-	t="${t%
-_}"
-	if [ 1 -gt $("${READTAGS}" -t "$tags" - "$t" | wc -l) ]; then
-	    printf 'FAILED: "%s" -t "%s" - "%s"\n' "${READTAGS}" "$tags" "$t"
-	    printf '	The raw tag name was "%s"\n' "$name"
-	    s=1
-	fi
+    for name in $(cut -f1 < "$tags" | limit_entries); do
+        if [ "$name" = "\r" ] ; then
+            t="$(printf "\r")"
+        elif [ "$name" = "\n" ] ; then
+            t="$IFS"
+        else
+            t="$(printf '%s\n' "$name" | expandEscapeSequences)"
+        fi
+
+        if [ 1 -gt $("${READTAGS}" -t "$tags" - "$t" | wc -l) ]; then
+            printf 'FAILED: "%s" -t "%s" - "%s"\n' "${READTAGS}" "$tags" "$t"
+            printf '	The raw tag name was "%s"\n' "$name"
+            s=1
+        fi
```

Comment:
L138-142: nothing to do except for indentation

Reply:
I think it is possible that you can remove L138-142 from the change by adjusting indentation (or whitespaces).
I think you can use tagnames, the original variables.
Does FreeBSD require removing the comment?
Updating the comment for FreeBSD is welcome. However, removing the whole the comment is not needed, I think.

The minimized version may be:
```
diff --git a/misc/roundtrip b/misc/roundtrip
index 536da3add..04006776b 100644
--- a/misc/roundtrip
+++ b/misc/roundtrip
@@ -126,7 +126,7 @@ IFS='
 
 tagfiles=$(find "$UNITS" -name expected.tags)
 for tags in $tagfiles; do
-    tagnames=$(sed -e 's/^\([^	]*\)	.*/\1/' "$tags" | limit_entries)
+    tagnames=$(cut -f1 < "$tags" | limit_entries)
     for name in $tagnames; do
 	# Yes, there is a reason for this craziness.  We need to properly
 	# handle embedded newlines (expanded from "\n"), including trailing
@@ -135,9 +135,13 @@ for tags in $tagfiles; do
 	# remove it, plus the extra newline, using variable substitutions.
 	# Note: we use "printf '%s\n'" instead of "echo" because Dash's "echo"
 	# unconditionally expands some sequences, like "\t" and alike.
-	t="$(printf '%s\n' "$name" | expandEscapeSequences; printf _)"
-	t="${t%
-_}"
+	if [ "$name" = "\r" ] ; then
+	    t="$(printf "\r")"
+	elif [ "$name" = "\n" ] ; then
+	    t="$IFS"
+	else
+	    t="$(printf '%s\n' "$name" | expandEscapeSequences)"
+	fi
 	if [ 1 -gt $("${READTAGS}" -t "$tags" - "$t" | wc -l) ]; then
 	    printf 'FAILED: "%s" -t "%s" - "%s"\n' "${READTAGS}" "$tags" "$t"
 	    printf '	The raw tag name was "%s"\n' "$name"

```
Could you make my differ smaller?

If you think we should remove the comment or remove the variable, of course, it is o.k. Improving the code are welcome.
However, they should be done in another commit with another reason.

Reply:
I don' known why should be done in another commit with another reason.

When I fixed this problem as you expected, the code will be very ugly and no any readability. 
ugly: no indentation and redundant variables
no any readability: the comments not match the code

In my idea, the code must be written for people to read, the readability is very important.

Reply:
> I don' known why should be done in another commit with another reason.

Keeping a commit for single purpose reduces the load of reviewers.
In this case, I am the reviewer.

The original commit does more than one thing:

* improving readability,
* making the script work on FreeBSD

Updating g the comment for the change making the script work on FreeBSD is acceptable.
I don't talk about the number of lines.

I found a better explanation what I would like to say in 
https://www.kernel.org/doc/html/v4.10/process/submitting-patches.html#separate-your-changes .

I know "keeping a commit for single purpose" requires extra work to the author of the changes.

Reply:
Thank you for explaining.